app/controllers/file_manager_controller.rb

```diff
@@ -0,0 +1,240 @@
+# Code for file manager adapted from: adrientoub/file-explorer
+require 'archive'
+require 'pathname'
+
+class FileManagerController < ApplicationController
+  BASE_DIRECTORY = Rails.root.join('courses')
+  skip_before_action :set_course
+  skip_before_action :authorize_user_for_course
+  skip_before_action :update_persistent_announcements
+
+  def index
+    check_path_exist('')
+    populate_directory(BASE_DIRECTORY, '')
+  end
+
+  def upload_index
+    upload_file('')
+    populate_directory(BASE_DIRECTORY, '')
+    render 'file_manager/index'
+  end
+
+  def path
+    absolute_path = check_path_exist(params[:path].nil? ? "" : params[:path])
+    if File.directory?(absolute_path) && check_instructor(absolute_path)
+      populate_directory(absolute_path, "#{params[:path]}/")
+      render 'file_manager/index'
+    elsif File.file?(absolute_path) && check_instructor(absolute_path)
+      if File.size(absolute_path) > 1_000_000 || params[:download]
+        send_file absolute_path
+      else
+        @file = File.read(absolute_path)
+        @path = params[:path]
+        @parent = './'
```

Comment:
This seems to be the only value of `@parent` used -- why not hardcode `./` directly in `file.html.erb`?

Reply:
Did this too.